modules/cicd/github-actions/main.tf

```diff
@@ -0,0 +1,193 @@
+data "aws_eks_cluster" "this" {
+  name = var.cluster_name
+}
+
+data "aws_region" "current" {}
+
+resource "kubernetes_namespace" "this" {
+  count = var.namespace == "" ? 1 - local.argocd_enabled : 0
+  metadata {
+    name = var.namespace_name
+  }
+}
+
+resource "local_file" "namespace" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content = yamlencode({
+    "apiVersion" = "v1"
+    "kind"       = "Namespace"
+    "metadata" = {
+      "name" = local.namespace
+    }
+  })
+  filename = "${path.root}/${var.argocd.path}/ns-${local.namespace}.yaml"
+}
+
+resource "local_file" "github_runners" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
```

Comment:
Terraform now supports `depends_on` for modules, is this still necessary?

Reply:
let's leave it as it is, we can pass an empty list as dependencies here, just for compatibility with terraform <12 and others modules